lib/logstash/pipeline.rb

```diff
@@ -59,14 +59,32 @@ def ready?
   end
 
   def configure(setting, value)
-    if setting == "filter-workers" && value > 1
-      # Abort if we have any filters that aren't threadsafe
-      plugins = @filters.select { |f| !f.threadsafe? }.collect { |f| f.class.config_name }
-      if !plugins.size.zero?
-        raise LogStash::ConfigurationError, "Cannot use more than 1 filter worker because the following plugins don't work with more than one worker: #{plugins.join(", ")}"
+    @settings[setting] = value
+  end
+
+  def safe_filter_worker_count
+    default = @settings["default-filter-workers"]
+    thread_count = @settings["filter-workers"] #override from args "-w 8" or config
+    safe_filters, unsafe_filters = @filters.partition(&:threadsafe?)
```

Comment:
I want to keep this if possible.  They are temporary arrays of pointers to filters instances.  We can perhaps look at the ratio of safe vs unsafe later and partition is an efficient way of splitting them. WDYT?